File: clsEnkripsi.vb
Option Strict On

Imports System.IO
Imports System.Security.Cryptography
Imports System.Text

Public Class Encryption64

    Public Shared Function Encrypt(ByVal stringToEncrypt As String, ByVal SEncryptionKey As String) As String
        Dim aReturn As String = String.Empty
        Dim key() As Byte = {}
        Dim IV() As Byte = {&H12, &H34, &H56, &H78, &H90, &HAB, &HCD, &HEF}
        Try
            key = System.Text.Encoding.UTF8.GetBytes(Left(SEncryptionKey, 8))
            Dim des As New DESCryptoServiceProvider()
            Dim inputByteArray() As Byte = Encoding.UTF8.GetBytes(stringToEncrypt)
            Dim ms As New MemoryStream()
            Dim cs As New CryptoStream(ms, des.CreateEncryptor(key, IV), CryptoStreamMode.Write)

            cs.Write(inputByteArray, 0, inputByteArray.Length)
            cs.FlushFinalBlock()
            aReturn = Convert.ToBase64String(ms.ToArray())
        Catch e As Exception
            aReturn = e.Message
        End Try
        Return aReturn

    End Function

    Public Shared Function Decrypt(ByVal stringToDecrypt As String,
        ByVal sEncryptionKey As String) As String
        Dim key() As Byte = {}
        Dim IV() As Byte = {&H12, &H34, &H56, &H78, &H90, &HAB, &HCD, &HEF}

        If stringToDecrypt = "" Then
            Return ""
        End If
        Dim inputByteArray(stringToDecrypt.Length) As Byte
        Try
            key = System.Text.Encoding.UTF8.GetBytes(Left(sEncryptionKey, 8))
            Dim des As New DESCryptoServiceProvider()
            inputByteArray = Convert.FromBase64String(stringToDecrypt)
            Dim ms As New MemoryStream()
            Dim cs As New CryptoStream(ms, des.CreateDecryptor(key, IV),
                CryptoStreamMode.Write)
            cs.Write(inputByteArray, 0, inputByteArray.Length)
            cs.FlushFinalBlock()
            Dim encoding As System.Text.Encoding = System.Text.Encoding.UTF8
            Return encoding.GetString(ms.ToArray())
        Catch e As Exception
            Return e.Message
        End Try
    End Function

End Class

File: jomiunsWords/clsEnkripsi.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;

public class Encryption64
{
    public static string Encrypt(string stringToEncrypt, string sEncryptionKey)
    {
        var aReturn = string.Empty;
        var key = Encoding.UTF8.GetBytes(sEncryptionKey.Length > 8 ? sEncryptionKey.Substring(0, 8) : sEncryptionKey.PadRight(8, '\0'));
        var IV = new byte[] { 0x12, 0x34, 0x56, 0x78, 0x90, 0xAB, 0xCD, 0xEF };
        using (var des = new DESCryptoServiceProvider())
        {
            var inputByteArray = Encoding.UTF8.GetBytes(stringToEncrypt);
            using (var ms = new MemoryStream())
            {
                using (var cs = new CryptoStream(ms, des.CreateEncryptor(key, IV), CryptoStreamMode.Write))
                {
                    cs.Write(inputByteArray, 0, inputByteArray.Length);
                    cs.FlushFinalBlock();
                    aReturn = Convert.ToBase64String(ms.ToArray());
                }
            }
        }
        return aReturn;
    }

    public static string Decrypt(string stringToDecrypt, string sEncryptionKey)
    {
        var IV = new byte[] { 0x12, 0x34, 0x56, 0x78, 0x90, 0xAB, 0xCD, 0xEF };

        if (string.IsNullOrEmpty(stringToDecrypt))
            return "";

        var key = Encoding.UTF8.GetBytes(sEncryptionKey.Length > 8 ? sEncryptionKey.Substring(0, 8) : sEncryptionKey.PadRight(8, '\0'));
        using (var des = new DESCryptoServiceProvider())
        {
            var inputByteArray = Convert.FromBase64String(stringToDecrypt);
            using (var ms = new MemoryStream())
            {
                using (var cs = new CryptoStream(ms, des.CreateDecryptor(key, IV), CryptoStreamMode.Write))
                {
                    cs.Write(inputByteArray, 0, inputByteArray.Length);
                    cs.FlushFinalBlock();
                    return Encoding.UTF8.GetString(ms.ToArray());
                }
            }
        }
    }
}